include/kllvm/util/temporary_file.h

```diff
@@ -0,0 +1,50 @@
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <unistd.h>
+
+typedef std::unique_ptr<FILE, decltype(&fclose)> temp_c_file_type;
```

Comment:
Another non-obvious, nuanced improvement to this code is to use a default-constructible functor type for the deleter, rather than `decltype(&fclose)`: https://stackoverflow.com/a/63942318